cli/src/pcluster/models/cluster.py

```diff
@@ -627,32 +639,14 @@ def get_running_capacity(self, updated_value: bool = False):
     def start(self):
         """Start the cluster."""
         try:
+            stack_status = self.stack.status
+            if "FAILED" in stack_status:
+                LOGGER.warning("Cluster stack is in %s status. This operation might fail.", stack_status)
             scheduler = self.config.scheduling.scheduler
             if scheduler == "awsbatch":
-                LOGGER.info("Enabling AWS Batch compute environment : %s", self.name)
-                try:
-                    compute_resource = self.config.scheduling.queues[0].compute_resources[0]
-
-                    AWSApi.instance().batch.enable_compute_environment(
-                        ce_name=self.stack.batch_compute_environment,
-                        min_vcpus=compute_resource.min_vcpus,
-                        max_vcpus=compute_resource.max_vcpus,
-                        desired_vcpus=compute_resource.desired_vcpus,
-                    )
-                except Exception as e:
-                    raise _cluster_error_mapper(e, f"Unable to enable Batch compute environment. {str(e)}")
-
+                self.enable_awsbatch_compute_environment()
             else:  # scheduler == "slurm"
-                stack_status = self.stack.status
-                if "IN_PROGRESS" in stack_status:
-                    raise ClusterActionError(f"Cannot start compute fleet while stack is in {stack_status} status.")
-                if "FAILED" in stack_status:
-                    LOGGER.warning("Cluster stack is in %s status. This operation might fail.", stack_status)
-
-                compute_fleet_status_manager = ComputeFleetStatusManager(self.name)
-                compute_fleet_status_manager.update_status(
-                    ComputeFleetStatus.START_REQUESTED, ComputeFleetStatus.STARTING, ComputeFleetStatus.RUNNING
-                )
+                self.start_slurm_compute_fleet()
         except ComputeFleetStatusManager.ConditionalStatusUpdateFailed:
             raise ClusterActionError(
```

Comment:
can this become a BadRequestClusterActionError?

Reply:
what are the errors being caught by?
```
except ClusterActionError as e:
            raise e
```

Reply:
Removed the 
```
except ClusterActionError as e:
            raise e
```